src/Microsoft.OData.Core/Json/IJsonWriterFromStreamFactory.cs

```diff
@@ -0,0 +1,27 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Microsoft.OData.Json
+{
+    /// <summary>
+    /// The interface of factories that create <see cref="IJsonWriter"/> instances
+    /// that write directly to a <see cref="Stream"/> rather than a <see cref="TextWriter"/>.
+    /// </summary>
+    [CLSCompliant(false)]
+    public interface IStreamBasedJsonWriterFactory
+    {
+        /// <summary>
+        /// Creates a new JSON writer of <see cref="IJsonWriter"/>. Returns null
+        /// if it cannot create a writer with the specified constraints.
+        /// </summary>
+        /// <param name="stream">Output stream to which data should be written.</param>
+        /// <param name="isIeee754Compatible">True if it is IEEE754Compatible.</param>
+        /// <param name="encoding">The text encoding of the output data.</param>
+        /// <returns>The JSON writer created.</returns>
+        IJsonWriter CreateJsonWriter(Stream stream, bool isIeee754Compatible, Encoding encoding);

```

Comment:
shall we accept a json writer settings from outside?

Reply:
I was not sure whether it was a good idea to expose `Utf8JsonWriter`-specific settings to the public API. I tried to keep the `Utf8JsonWriter` as an implementation detail. What are your thoughts?

Reply:
Since we are changing the `JsonWriterOptions` in the underlying implementation, I think it is more confusing than helpful to expose it as a parameter.